lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.AirLoopHVAC.rb

```diff
@@ -5,9 +5,135 @@ class ASHRAE901PRM < Standard
+
+    allowable_fan_bhp = 0.0
+    allowable_power_w = 0.0
+    if system_type == 'PSZ_AC' ||
+       system_type == 'PSZ_HP' ||
+       system_type == 'PVAV_Reheat'
+      system_type == 'PVAV_PFP_Boxes' ||
+        system_type == 'VAV_Reheat' ||
+        system_type == 'VAV_PFP_Boxes' ||
+        system_type == 'SZ_VAV' ||
+        system_type == 'SZ_CAV'
+
+      # Calculate the allowable fan motor bhp for the air loop
+      allowable_fan_bhp = air_loop_hvac_allowable_system_brake_horsepower(air_loop_hvac)
+
+      # Divide the allowable power evenly between the fans
+      # on this air loop.
+      # TODO: same proportions as proposed ?
+      allowable_fan_bhp /= all_fans.size
+    elsif system_type == 'PTAC' ||
+          system_type == 'PTHP' ||
+          (system_type == 'Gas_Furnace' && !is_nmc) ||
+          (system_type == 'Electric_Furnace' && !is_nmc)
+      # Determine allowable fan power
+      allowable_power_w = 0.3
+    elsif (system_type == 'Gas_Furnace' && is_nmc) ||
+          (system_type == 'Electric_Furnace' && is_nmc)
+      # Determine allowable fan power
+      allowable_power_w = 0.054
+    end
+
+    # Set the motor efficiencies
+    # for all fans based on the calculated
+    # allowed brake hp.  Then calculate the allowable
+    # fan power for each fan and adjust
+    # the fan pressure rise accordingly
+    all_fans.each do |fan|
+      if allowable_fan_bhp > 0
+        fan_apply_standard_minimum_motor_efficiency(fan, allowable_fan_bhp)
+        allowable_power_w = allowable_fan_bhp * 746 / fan.motorEfficiency
+      end
+      fan_adjust_pressure_rise_to_meet_fan_power(fan, allowable_power_w)
+    end
+
+    return true unless system_type == 'PVAV_PFP_Boxes' || system_type == 'VAV_PFP_Boxes'
+
+    # Adjust fan powered terminal fans power
+    air_loop_hvac.demandComponents.each do |dc|
+      next if dc.to_AirTerminalSingleDuctParallelPIUReheat.empty?
+
+      pfp_term = dc.to_AirTerminalSingleDuctParallelPIUReheat.get
+      air_terminal_single_duct_parallel_piu_reheat_apply_prm_baseline_fan_power(pfp_term)
+    end
+
+    return true
+  end
+
+  # Determine the allowable fan system brake horsepower
+  # Per Section G3.1.2.9
+  #
+  # @return [Double] allowable fan system brake horsepower
+  #   units = horsepower
+  def air_loop_hvac_allowable_system_brake_horsepower(air_loop_hvac)
+    # Get design supply air flow rate (whether autosized or hard-sized)
+    dsn_air_flow_m3_per_s = 0
+    dsn_air_flow_cfm = 0
+    if air_loop_hvac.autosizedDesignSupplyAirFlowRate.is_initialized
+      dsn_air_flow_m3_per_s = air_loop_hvac.autosizedDesignSupplyAirFlowRate.get
+      dsn_air_flow_cfm = OpenStudio.convert(dsn_air_flow_m3_per_s, 'm^3/s', 'cfm').get
+      OpenStudio.logFree(OpenStudio::Debug, 'openstudio.ashrae_90_1_prm.AirLoopHVAC', "* #{dsn_air_flow_cfm.round} cfm = Autosized Design Supply Air Flow Rate.")
+    else
+      puts air_loop_hvac.is_initialized
```

Comment:
Why is this "puts" statement here?

Reply:
Good catch, not sure what it's doing there. Probably a left over from testing I was doing before having access to Ruby debugger. I removed it in the last commit.